### PR TITLE
reports: Add palash25.md

### PR DESCRIPTION
**Pull request by palash25**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: _reports/palash25.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+led to a successful integration.
+
+#### Work to be done
+
+Two major features were left out because of time constraints:
+1. Ignore directories functionality for the NextGen-Core.
+2. Cache control flags for the NextGen-Core (which was
+   already mentioned as a stretch issue in my GSoC proposal).
+
+I have started the work on implementing the cache control flags and will
```
Comment (jayvdb) [inline _reports/palash25.md]: remove this paragraph.

Work to be done is not about *you* doing the work; it is just stuff which is left over.

While we hope you'll continue, this work report is the end of your commitment.

File: _reports/palash25.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  - 2018
+  - StatusReport
+report_year: 2018
+date: 2018-08-12
+student: Palash Nigam
+organisation: coala
+organisation_link: https://coala.io
+project: Optimize Caching for the NextGen-Core
+project_link: https://summerofcode.withgoogle.com/projects/#6434190552203264
+tarball:
```
Comment (jayvdb) [inline _reports/palash25.md]: missing ;-)

File: _reports/palash25.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      - details: >
+         Processing: Add middleware `FileDict` that provides the bears
+         with the actual file contents instead of the `FileFactory` objects.
+  - 14:
+      - repo: coala
+      - link: https://github.com/coala/coala/commit/e138e4875fc6d1474a93758881f550e05df8e329
+      - details: >
+         Processing: Modify tests to use actual files instead of hard coded
+         tuples as file content.
+  - 15:
```
[the remark this answers is not in the document]
Comment (palash25) [reply _reports/palash25.md]: Sorry I forgot the cEP patch

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/635/commits/dcc00df79bb3264bb930c3ae015104ffd09d1187

**Comment by palash25** [comment]
all green :tada:

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**